changes/tasks.py

```diff
@@ -1468,6 +1468,88 @@ def run(self, vumi_message_id: str, event_type: str, **kwargs) -> None:
 process_whatsapp_system_event = ProcessWhatsAppSystemEvent()
 
 
+class ProcessWhatsAppTimeoutSystemEvent(Task):
+    """
+    Notifies a user that a message we send them on Whatsapp has not been delivered
+    within 7 days, she will be sent an outbound
+    message informing them and give the option to switch to SMS.
+    """
+
+    name = "ndoh_hub.changes.tasks.process_whatsapp_timeout_system_event"
+
+    def handle_undelivered(self, user_id, source_id, identity_uuid):
+        identity = is_client.get_identity(identity_uuid)
+        details = identity["details"]
+        if "timeout_timestamp" not in details:
+            identity["details"]["timeout_timestamp"] = datetime.today()
```

Comment:
Just setting it here doesn't store it in the identity store, we need to call the `is_client.update_identity` to ensure that this value is stored.